### ISSUE TITLE
Updated month prop not updating the calendar when displaying multiple months

### ISSUE THREAD

**Issue by geminiyellow** [gpbl/react-day-picker#580]
`6.x` it work very well, but when i update to `7.x`, the `month` cannot causes the calendar to re-render when its value changes.

i save the `month` in component's state, and it looks like: http://react-day-picker.js.org/examples/elements-year-navigation

but i set the `numberOfMonths` to `2`. and then setState from `captionElement`.
the logs:

![image](https://user-images.githubusercontent.com/929878/33541347-32c831a4-d912-11e7-9230-aa2d78689bf0.png)

in `handleChange` i got the right date, `201706`, then setState to component, 
then the `captionElement` got the `201711` and `201712`.

then i try to change not use state cause picker rerender, change to `showMonth`.
now it work. 

ok, then i read the docs, and try to use `onMonthChange` to keep the `currentMonth` to the state, but WTF, you see that? `captionElement` use the old values. 

there is no BREAKINGCHANGE in the CHANGELOG and there is no docs in anywhere. 
so , i cannot use 7.x. @gpbl could you help me ?

my code:

![image](https://user-images.githubusercontent.com/929878/33541824-f385f042-d913-11e7-8349-0bb01cba00c1.png)

**Comment by geminiyellow** [comment]
ok, now i can say that is a bug.

from here: [Fixed render of months when toMonth equals current month](https://github.com/gpbl/react-day-picker/commit/355f86148bd1c276019137d0d27ba7b5f848e720)

@hydrognomik add this

```diff
+    } else if (props.toMonth && props.numberOfMonths > 1) {
 +      const diffInMonths = Helpers.getMonthsDiff(props.toMonth, currentMonth);
 +      if (diffInMonths <= 0) {
 +        currentMonth = DateUtils.addMonths(
 +          props.toMonth,
 +          1 - this.props.numberOfMonths
 +        );
 +      }
```

`Helpers.getMonthsDiff(props.toMonth, currentMonth)` my `toMonth` is `201712`, `currentMonth` is `201709`, then the `diffInMonths` is `-3`, why you need to reset the `currentMonth` ? that is not right.

you should swap it to  `Helpers.getMonthsDiff(currentMonth, props.toMonth)`

**Comment by gpbl** [comment]
Thanks @geminiyellow ! Thanks for finding out the bug and the solution 👍 Going to release a fix ASAP

**Comment by geminiyellow** [comment]
wow thank you @gpbl

**Comment by gpbl** [comment]
Fix published in `v7.0.5`!
